File: data_structures/cpp/src/04_selection_sort.cpp
// Selection Sort Algorithm

#include <iostream>

using namespace std;

void display_array(int array[], int max_size)
{
    for (int i = 0; i < max_size; i++)
    {
        cout << array[i] << " ";
    }
    cout << endl;
}

void selection_sort(int array[], int max_size)
{
    for (int i = 0; i < max_size; i++)
    {
        int min_index = i;
        for (int j = i + 1; j < max_size; j++)
        {
            if (array[j] < array[i])
            {
                min_index = j;
            }
        }
        int temp = array[i];
        array[i] = array[min_index];
        array[min_index] = temp;
    }
    display_array(array, max_size);
}

int main()
{
    int array[] = {10, 9, 8, 7, 6, 5, 4, 3, 2, 1};
    int max_size = sizeof(array) / sizeof(array[0]);
    selection_sort(array, max_size);
    return 0;
}

File: python_data_structures/04_selection_sort.py
# Selection Sort Algorithm


def selection_sort(array):
    array_size = len(array)

    index = 0
    while index < array_size:
        min_index = index
        for i in range(index + 1, array_size):
            if array[i] < array[index]:
                min_index = i

        array[index], array[min_index] = array[min_index], array[index]
        index += 1

    return array


if __name__ == "__main__":
    array = [10, 9, 8, 7, 6, 5, 4, 3, 2, 1]
    sorted_array = selection_sort(array)
    print(sorted_array)
